components/permalink_redirector/actions.js

```diff
@@ -0,0 +1,48 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {getChannel} from 'mattermost-redux/actions/channels';
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+import {getTeam as getTeamSelector} from 'mattermost-redux/selectors/entities/teams';
+import {getPost} from 'mattermost-redux/actions/posts';
+import {getTeam} from 'mattermost-redux/actions/teams';
+
+import {redirectUserToDefaultTeam} from 'actions/global_actions.jsx';
+
+import {Constants, ErrorPageTypes} from 'utils/constants.jsx';
+import {browserHistory} from 'utils/browser_history';
+
+import LocalStorageStore from 'stores/local_storage_store';
+
+export const redirect = (postId) => {
+    return async (innerDispatch, getState) => {
```

Comment:
nit: Is it actually necessary to name this `innerDispatch` vs. just `dispatch`?

Reply:
refactor kitty has bitten me. :)